src/cephadm/cephadmlib/systemd_unit.py

```diff
@@ -2,22 +2,138 @@
 
 import contextlib
 import os
+import json
 import pathlib
 
-from typing import IO, List, Optional, Union
+from typing import Any, Dict, IO, List, Optional, Union
 
 from . import templating
 from .call_wrappers import call_throws
 from .container_engines import Docker, Podman
 from .context import CephadmContext
 from .daemon_identity import DaemonIdentity, DaemonSubIdentity
+from .exceptions import SystemdDropInException
 from .file_utils import write_new
 from .logging import write_cluster_logrotate_config
 
 
 _DROP_IN_FILENAME = '99-cephadm.conf'
 
 
+class SystemdDropIn:
+    """Class for specifying the contents of a systemd drop-in file"""
+
+    def __init__(self) -> None:
+        self.contents: Dict[str, Dict[str, List[str]]] = {}
+
+    @classmethod
+    def from_dict(
+        self, data: Dict[str, Dict[str, List[str]]]
+    ) -> 'SystemdDropIn':
+        dropin = SystemdDropIn()
+        dropin.contents = data
+        return dropin
+
+    def add_section(self, section: str) -> None:
+        if section not in self.contents.keys():
+            self.contents[section] = {}
+
+    def add_setting(self, section: str, setting: str, value: str) -> None:
```

Comment:
This is all very similar to configparser. systemd unit format is extremely similar to python ini too. Should we perhaps make them compatible by using the same/similar method names?